cypress/e2e/universal-search.test.ts

```diff
@@ -35,6 +35,32 @@ describe('Universal search bar', () => {
     cy.contains('UNI is the governance token for Uniswap').should('exist')
   })
 
+  it('should go to the selected result when recent results are shown', () => {
+    // Search for uni token by name.
+    cy.get('[data-cy="search-bar-input"]').last().type('uni')
+    cy.get('[data-cy="searchbar-token-row-UNI"]')
+
+    // Clear search
+    cy.get('[data-cy="search-bar-input"]').last().clear()
+
+    // Close search
+    cy.get('[data-cy="search-bar-input"]').last().type('{esc}')
+
+    openSearch()
+
+    // Search a different token by name.
+    cy.get('[data-cy="search-bar-input"]').last().type('eth')
+
+    // Validate ETH result now exists.
+    cy.get('[data-cy="searchbar-token-row-ETH"]')
+
+    // Hit enter
+    cy.get('[data-cy="search-bar-input"]').last().type('{enter}')
+
+    // Validate we went to ethereum address
+    cy.url().should('contain', 'tokens/ethereum/NATIVE')
```

Comment:
The search bar element should be static (I think), so you can grab it once and re-use it from a closure:

```
    cy.get('[data-cy="search-bar-input"]').last().then((searchBar) => {
      // Search for uni token by name.
      cy.wrap(searchBar).type('uni')
      ...
    })
```

Also, avoid new instances of `data-cy`, using `data-testid` (and `getTestSelector`) instead. No changes necessary for this (unless you want to), just something to keep in mind.

Reply:
good points, addressed the first but have a lot of unblocking work to get to so i'll remember to avoid data-cy in future pr's!